Modix.Services/Mentions/MentionService.cs

```diff
@@ -11,53 +11,65 @@ namespace Modix.Services.Mentions
     public interface IMentionService
     {
         /// <summary>
-        /// Determines whether the user can mention the supplied role.
+        /// Mentions the given role in the given channel
         /// </summary>
-        /// <param name="role">The role that the user is trying to mention.</param>
-        /// <exception cref="ArgumentNullException">Throws for <paramref name="role"/>.</exception>
-        void AuthorizeMention(IRole role);
-
-        /// <summary>
-        /// Ensures that the role can be mentioned.
-        /// </summary>
-        /// <param name="role">The role to be mentioned.</param>
-        /// <returns>
-        /// A <see cref="Task"/> that will complete when the operation completes,
-        /// with a delegate that can be invoked to restore the role to its previous configuration.
-        /// </returns>
-        Task<Func<Task>> EnsureMentionableAsync(IRole role);
+        /// <param name="role">The role to mention</param>
+        /// <param name="channel">The channel to mention in</param>
+        Task MentionRoleAsync(IRole role, IMessageChannel channel);
     }
 
     /// <inheritdoc />
     internal class MentionService : IMentionService
     {
         public MentionService(
             IDiscordClient discordClient,
-            IAuthorizationService authorizationService)
+            IAuthorizationService authorizationService,
+            IDesignatedRoleService designatedRoleService)
         {
             DiscordClient = discordClient;
             AuthorizationService = authorizationService;
+            DesignatedRoleService = designatedRoleService;
         }
 
-        /// <inheritdoc />
-        public void AuthorizeMention(IRole role)
+        private void AuthorizeMention(IRole role)
         {
-            AuthorizationService.RequireClaims(AuthorizationClaim.MentionRestrictedRole);
-
             if (role is null)
                 throw new ArgumentNullException(nameof(role));
+
+            AuthorizationService.RequireClaims(AuthorizationClaim.MentionRestrictedRole);
         }
 
-        public async Task<Func<Task>> EnsureMentionableAsync(IRole role)
+        /// <inheritdoc />
+        public async Task MentionRoleAsync(IRole role, IMessageChannel channel)
         {
-            if (!role.IsMentionable)
+            if (role.IsMentionable)
```

Comment:
This occurs before the null check now. Honestly, it's probably better to just get rid of the AuthorizeMention method entirely, put the null check above this, and put the RequireClaims call where the AuthorizeMention call currently is